x-pack/plugins/code/common/repository_utils.ts

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import GitUrlParse from 'git-url-parse';
+import path from 'path';
+import { Location } from 'vscode-languageserver';
+
+import { CloneProgress, FileTree, FileTreeItemType, Repository, RepositoryUri } from '../model';
+import { parseLspUrl, toCanonicalUrl } from './uri_util';
+
+export class RepositoryUtils {
+  // Generate a Repository instance by parsing repository remote url
+  public static buildRepository(remoteUrl: string): Repository {
+    const repo = GitUrlParse(remoteUrl);
+    let host = repo.source ? repo.source : '';
+    if (repo.port !== null) {
+      host = host + ':' + repo.port;
+    }
+    const name = repo.name ? repo.name : '_';
+    const org = repo.owner ? repo.owner.split('/').join('_') : '_';
+    const uri: RepositoryUri = host ? `${host}/${org}/${name}` : repo.full_name;
+    return {
+      uri,
+      url: repo.href as string,
+      name,
+      org,
+    };
+  }
+
+  // From uri 'origin/org/name' to 'org'
+  public static orgNameFromUri(repoUri: RepositoryUri): string {
+    const segs = repoUri.split('/');
+    if (segs && segs.length === 3) {
+      return segs[1];
+    } else {
+      return 'invalid';
```

Comment:
Returning `'invalid'` feels like a good way to hide errors from invalid input, what would be wrong with throwing an error in the case that invalid input was received?

Reply:
done